cypress_test/integration_tests/desktop/writer/top_toolbar_spec.js

```diff
@@ -330,4 +330,85 @@ describe('Top toolbar tests.', function() {
 			.should('exist');
 	});
 
+	it('Save.', function() {
+		cy.get('#tb_editbar_item_bold')
+			.click();
+
+		cy.get('#tb_editbar_item_save')
+			.click();
+
+		helper.beforeAll(testFileName, 'writer', true);
+
+		helper.reselectAllText();
+
+		cy.get('#copy-paste-container p b')
+			.should('exist');
+
+	});
+
+	it('Print', function() {
+		// A new window should be opened with the PDF.
+		cy.window()
+			.then(function(win) {
+				cy.stub(win, 'open');
+			});
+
+		cy.get('#tb_editbar_item_print')
+		    .click();
+
+		cy.window().its('open').should('be.called');
+	});
+	
+	it('Apply Undo/Redo.', function() {
+		cy.get('#tb_editbar_item_italic')
+			.click();
+
+		helper.selectAllText(false);
+
+		cy.get('#copy-paste-container p i')
+			.should('exist');
+		helper.selectAllText(false);
+
+		//Undo
+		cy.get('#tb_editbar_item_undo')
+			.click();
+
+		helper.selectAllText(false);
+
+		cy.get('#copy-paste-container p i')
+			.should('not.exist');
+
+		//Redo
+		cy.get('#tb_editbar_item_redo')
+			.click();
+
+		helper.selectAllText(false);
+
+		cy.get('#copy-paste-container p i')
+			.should('exist');
+	});
+
+	it('Hide/Show sidebar.', function() {
```

Comment:
About showing/hiding sidebar we already have some code. Search for showSidebar(), hideSidebar() methods. You should use those methods directly here, to avoid code duplication. We use those helper methods in other tests, but it's good to have show/hide sidebar test cases covered here alone.